ledger-core/v2/insolar/runner/runner.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/insolar/blob/master/LICENSE.md.
+
+package runner
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/record"
+	"github.com/insolar/assured-ledger/ledger-core/v2/logicrunner/common"
+	"github.com/insolar/assured-ledger/ledger-core/v2/runner/requestresult"
+)
+
+type ContractExecutionStateUpdateType int
+
+const (
+	_ ContractExecutionStateUpdateType = iota
+	ContractError
+	ContractAborted
+	ContractDone
+
+	ContractOutgoingCall
+)
+
+type RPCEvent interface{}
+
+type ContractExecutionStateUpdate struct {
+	Type  ContractExecutionStateUpdateType
+	Error error
+
+	Result   *requestresult.RequestResult
+	Outgoing RPCEvent
+}
+
+type Execution struct {
+	Object     insolar.ObjectDescriptor
+	Context    context.Context
+	Request    *record.IncomingRequest
+	Nonce      int64
+	Deactivate bool
+	Pulse      insolar.Pulse
+
+	LogicContext insolar.LogicCallContext
+}
+
+type Runner interface {
```

Comment:
I'd rather call this ContractExecutor, and virtual.Virtual could be virtual.Runner, so there will be virtual.NewRunner instead of virtual.NewVirtual

Reply:
Don't get it. Rename interface Runner to ContractExecutor? Why virtual.Virtual should be virtual.Runner?

Reply:
disregard my comment :)